### PR TITLE
Enable iree_static_linker_test for cross compile

### PR DESCRIPTION
**Pull request by hcindyl**
Enable cross compile flags in `iree_bytecode_module` and `iree_c_module`, and allow `iree_static_linker_test` to build and test in cross compile platforms.

### CODE REVIEW

File: build_tools/cmake/iree_bytecode_module.cmake [hunk 1]
```diff
@@ -109,6 +109,20 @@ function(iree_bytecode_module)
...
+    # https://developer.android.com/ndk/guides/cmake#android_platform
+    #
+    # In typical CMake fashion, the various strings are pretty fuzzy and can
+    # have multiple values like "latest", "android-25"/"25"/"android-N-MR1".
+    #
+    # From looking at the toolchain file, ANDROID_PLATFORM_LEVEL seems like it
+    # should pretty consistently be just a number we can use for target triple.
+    set(_TARGET_TRIPLE "aarch64-none-linux-android${ANDROID_PLATFORM_LEVEL}")
+    list(APPEND _ARGS "--iree-llvm-target-triple=${_TARGET_TRIPLE}")
+  endif()
```
Comment (ScottTodd) [inline build_tools/cmake/iree_bytecode_module.cmake:124]: Ehhhh... this should be reconciled with https://github.com/iree-org/iree/blob/8ebe002ebebf9e2f867cc808d8a11e8658ea9ea7/build_tools/cmake/iree_check_test.cmake#L14-L43 somehow (I don't like that `iree_bytecode_module_for_iree_check_test_and_friends` exists in the first place)

Either `iree_bytecode_module` should learn all these tricks or it should know none of them, and `iree_static_linker_test` should use `iree_bytecode_module_for_iree_check_test_and_friends` instead :/

File: build_tools/cmake/iree_c_module.cmake [hunk 2]
```diff
@@ -81,6 +81,36 @@ function(iree_c_module)
...
+    #
+    # From looking at the toolchain file, ANDROID_PLATFORM_LEVEL seems like it
+    # should pretty consistently be just a number we can use for target triple.
+    set(_TARGET_TRIPLE "aarch64-none-linux-android${ANDROID_PLATFORM_LEVEL}")
+    list(APPEND _ARGS "--iree-llvm-target-triple=${_TARGET_TRIPLE}")
+  endif()
+
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+     CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+     NOT _RULE_FLAGS MATCHES "iree-llvm-target-triple")
```
Comment (ScottTodd) [inline build_tools/cmake/iree_c_module.cmake:100]: Let's not duplicate this yet again. Time for a helper function?

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: build_tools/cmake/iree_c_module.cmake (hunk 2 above)
Comment (ScottTodd) [reply build_tools/cmake/iree_c_module.cmake:100]: (`iree_infer_flags_for_platform()` or something?)

Comment (hcindyl) [reply build_tools/cmake/iree_c_module.cmake:100]: yeah, I will move the helper function to `iree_macros.cmake` Thanks for the suggestion.

File: build_tools/cmake/iree_bytecode_module.cmake (hunk 1 above)
Comment (hcindyl) [reply build_tools/cmake/iree_bytecode_module.cmake]: There are other checks in `iree_bytecode_module_for_iree_check_test_and_friends` regarding cpu features which are not easily consolidated with the underlying `iree_bytecode_module`. At the same time, the static lib support doesn't need to be bubbled one level up.

I consolidate the platform-specific flag assignment in all files, but I would keep the  `iree_bytecode_module_for_iree_check_test_and_friends`  for now.

**Review by ScottTodd — APPROVED** [review]
Great, thanks!

File: build_tools/cmake/iree_bytecode_module.cmake (hunk 1 above)
Comment (ScottTodd) [reply build_tools/cmake/iree_bytecode_module.cmake]: SGTM. Thanks for looking!

**Pull request merged**